### PR TITLE
[config] move config.Chain to blockchain package

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
as title

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: blockchain/blockchain.go [hunk 1]
```diff
@@ -26,12 +27,16 @@ import (
 	"github.com/iotexproject/iotex-core/blockchain/blockdao"
 	"github.com/iotexproject/iotex-core/blockchain/filedao"
 	"github.com/iotexproject/iotex-core/blockchain/genesis"
-	"github.com/iotexproject/iotex-core/config"
```
Comment (dustinxie) [inline blockchain/blockchain.go:29]: this is the blocking dependency we try to get rid of

File: config/config.go [hunk 2]
```diff
@@ -456,70 +374,6 @@ func NewSub(configPaths []string, validates ...Validate) (Config, error) {
...
-	case SigP256k1:
-		sk, _ := crypto.GenerateKey()
-		return sk.HexString()
-	case SigP256sm2:
-		sk, _ := crypto.GenerateKeySm2()
-		return sk.HexString()
-	}
-	return ""
-}
-
```
Comment (dustinxie) [inline config/config.go:522]: moved to `blockchain/config.go`

File: blockchain/blockchain.go [hunk 3]
```diff
@@ -97,13 +103,43 @@ type (
...
+		clk            clock.Clock
+		pubSubManager  PubSubManager
+		timerFactory   *prometheustimer.TimerFactory
+
+		// used by account-based model
+		bbf BlockBuilderFactory
+	}
 )
 
+func generateRandomKey(scheme string) string {
```
Comment (Liuhaai) [inline blockchain/blockchain.go:130]: it's only used in `config.go`, so it shouldn't be placed here

File: blockchain/blockchain.go [hunk 4]
```diff
@@ -119,60 +155,56 @@ func Productivity(bc Blockchain, startHeight uint64, endHeight uint64) (map[stri
...
+		clk:     clock.New(),
 	}
 	for _, opt := range opts {
-		if err := opt(chain, cfg); err != nil {
+		if err := opt(chain); err != nil {
 			log.S().Panicf("Failed to execute blockchain creation option %p: %v", opt, err)
 		}
 	}
+	if chain.pubSubManager == nil {
+		chain.pubSubManager = NewPubSub(200)
```
Comment (Liuhaai) [inline blockchain/blockchain.go:201]: have pubsubManager be `nil` if no option is passed?

Comment (dustinxie) [reply blockchain/blockchain.go:201]: right, if no option, it's `nil`, so create according to default value

**Review by millken — APPROVED** [review]

File: blockchain/blockchain.go (hunk 4 above)
Comment (Liuhaai) [reply blockchain/blockchain.go:187]: After offline discussion, the buffersize of pubManager is added into the config of blockchain.

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**